src/main/java/org/query/util/Utilities.java

```
@@ -1,5 +1,22 @@
 package org.query.util;
+import java.math.BigInteger;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
 
 public class Utilities {
-  // Not sure we need this anymore. Keeping in case something comes up
+  // Simple hashing using SHA-512
+  public static String hashString(String input) {
+    try {
+      MessageDigest md = MessageDigest.getInstance("SHA-512");
+      byte[] messageDigest = md.digest(input.getBytes());
+      BigInteger no = new BigInteger(1, messageDigest);
+      String hashtext = no.toString(16);
+      while (hashtext.length() < 32) {
+        hashtext = "0" + hashtext;
+      }
+      return hashtext;
+    } catch (NoSuchAlgorithmException e) {
+      throw new RuntimeException(e);
```

This should probably throw an internal exception so something useful is shown to the client.

---

That sounds good. Will change.